runelite-mixins/src/main/java/net/runelite/mixins/RSSpritePixelsMixin.java

```diff
@@ -35,12 +36,32 @@
 	@Inject
 	@Override
 	public BufferedImage toBufferedImage()
+	{
+		BufferedImage img = new BufferedImage(getWidth(), getHeight(), BufferedImage.TYPE_INT_ARGB);
+
+		try
+		{
+			toBufferedImage(img);
+		}
+		catch (SizeMismatchException e)
+		{
+			throw new RuntimeException(e);
+		}
+
+		return img;
+	}
+
+	@Inject
+	@Override
+	public void toBufferedImage(BufferedImage img) throws SizeMismatchException
 	{
 		int width = getWidth();
 		int height = getHeight();
+
+		SizeMismatchException.check(img.getWidth(), width, img.getHeight(), height);
```

Comment:
I would probably not bother with an exception dedicated to this since they all get wrapped in runtime exception anyway.